modules/tobira/src/main/java/org/opencastproject/tobira/impl/TobiraEndpoint.java

```diff
@@ -225,6 +250,69 @@ public Response harvest(
     }
   }
 
+  @GET
+  @Path("/callback/{token}")
+  @Produces(APPLICATION_JSON)
+  @RestQuery(
+      name = "callback",
+      description = "Auth callback API to get user information. This is used by Tobira to get user information,"
+          + "currently only the REMOTE_USER header is supported.",
+      pathParameters = {
+          @RestParameter(
+              name = "token",
+              isRequired = true,
+              description = "The token to authorize the request.",
+              type = Type.STRING
+          )
+      },
+      responses = {
+          @RestResponse(description = "User Outcome Data", responseCode = HttpServletResponse.SC_OK)
+      },
+      returnDescription = "Returns user information"
+  )
+  public Response callback(@PathParam("token") String token, @Context HttpHeaders headers) {
+
+    if (callbackToken == null || !callbackToken.equals(token)) {
+      return badRequest("Invalid token or callback disabled");
+    }
+
+    List<String> username = headers.getRequestHeader("REMOTE_USER");
```

Comment:
After much faffing about, I *finally* convinced my nginx reverse proxy to pass this header.

From my googling, underscores are either not allowed, or at the very least very much frowned upon in header names.  Can we make this REMOTE-USER, or something else?

Reply:
Considering this is in the tobira endpoint space I don't know if thats worth it.

If you want to move this to a more general endpoint space then sure.

Reply:
If you use Shibboleth SP with Apache httpd, the REMOTE_USER header would work out of the box.

Reply:
> If you use Shibboleth SP with Apache httpd, the REMOTE_USER header would work out of the box.

@geichelberger are you sure this is a header? [The Shibboleth documentation](https://shibboleth.atlassian.net/wiki/spaces/SP3/pages/2065335257/AttributeAccess#REMOTE_USER) explicitly states that this is not a header but an env variable from the CGI days.